pages/index.tsx

```diff
@@ -1,11 +1,51 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
 import { HomeLayout } from "../components/layout";
 import { SubmissionsTable } from "../components/submissions";
+import { Pagination } from "../interfaces";
 
 export default function HomePage() {
+  const [submissions, setSubmissions] = useState([]);
+  const [pagination, setPagination] = useState({} as Pagination);
```

Comment:
casting empty objects as something they're not is always dangerous. I assume Pagination type if not just an empty object correct? In which case you either need to have a proper initial value for pagination here or put in a null and tell the useState that it can be either: ```useState<Pagination | null>(null)```

I leave it up to you to decide which option is best in this case